lib/Navigation/Builder.php

```diff
@@ -166,22 +167,15 @@ public function getNavigation($activeDocument = null, $navigationRootDocument =
             // we don't have an active document, so we try to build the trail on our own
             $allPages = $navigation->findAllBy('uri', '/.*/', true);
 
-            /** @var Page|Page\Document $page */
             foreach ($allPages as $page) {
                 $activeTrail = false;
 
-                if ($activeDocument instanceof Document) {
-                    if ($page->getUri() && strpos($activeDocument->getRealFullPath(), $page->getUri() . '/') === 0) {
+                if ($activeDocument instanceof Document && $page instanceof Url && $page->getUri()) {
+                    if (strpos($activeDocument->getRealFullPath(), $page->getUri() . '/') === 0) {
                         $activeTrail = true;
                     }
-
-                    if ($page instanceof DocumentPage) {
-                        if ($page->getDocumentType() == 'link') {
-                            if ($page->getUri() && strpos($activeDocument->getFullPath(),
-                                    $page->getUri() . '/') === 0) {
-                                $activeTrail = true;
-                            }
-                        }
+                    if (strpos($activeDocument->getFullPath(), $page->getUri() . '/') === 0) {
```

Comment:
Where is `if ($page instanceof DocumentPage) {` and `if ($page->getDocumentType() == 'link') {` gone? That's still necessary in my opinion.

Reply:
@brusch Is in Line 173 (Url is a Subtype of DocumentPage and should always Type 'link', right?), otherwise the getUri method doesn't exist:
`if ($activeDocument instanceof Document && $page instanceof Url && $page->getUri()) {`

Reply:
https://github.com/pimcore/pimcore/pull/7487/commits/1bab47de457d980e81bf71a326862216b6ae506c

Reply:
👌 Thanks!